packages/subscription-manager/src/components/SubscriptionManagerContainer/SubscriptionManagerContainer.tsx

```diff
@@ -12,6 +13,43 @@ export type SubscriptionManagerContainerProps = {
 	children?: React.ReactNode;
 };
 
+const getEmailAddress = () => {
+	const subkey = getSubkey();
+
+	if ( ! subkey ) {
+		return null;
+	}
+
+	const decodedSubkeyValue = decodeURIComponent( subkey );
+
+	const firstPeriodIndex = decodedSubkeyValue.indexOf( '.' );
+	if ( firstPeriodIndex === -1 ) {
+		return null;
+	}
+
+	const emailAddress = decodedSubkeyValue.slice( firstPeriodIndex + 1 );
+	return emailAddress;
+};
+
+const getSubHeaderText = ( translate: LocalizeProps[ 'translate' ] ) => {
+	const emailAddress = getEmailAddress();
```

Comment:
This one could also be converted to a hook, in this case it would be able to call both `useSubscriberEmailAddress` and `useTranslate` (so no need to pass translate as a parameter). It could take advantage of useMemo as well, listening to changes on emailAddress.

Reply:
Good point!

I added some additional thoughts on an `useExternalUser` hook [here](https://github.com/Automattic/wp-calypso/pull/74861#issuecomment-1482360662)

Reply:
Thanks for the suggestions folks!

> [...] [refactor](https://github.com/Automattic/wp-calypso/pull/74733) will bring in a SubscriptionManagementContext which we can use to store application state. I suggest we create a hook that returns the email address together with a verified key, which by default is false, but switches to true after the first API request succeeds.

Since the refactor is not ready yet, I would convert both `getSubHeaderText` and `getEmailAddress` into hooks - so we have the email address display ready for the Monday launch and then this logic can be later replaced by the data from the context (once it is ready). How does that sound to you?

Reply:
Both functions are now hooks.